app/src/main/java/com/example/playlistmaker/AudioPlayerActivity.kt

```diff
@@ -1,50 +1,116 @@
 package com.example.playlistmaker
 
+import android.icu.text.SimpleDateFormat
 import android.os.Bundle
-import android.widget.Button
-import android.widget.ImageView
+import android.os.Handler
+import android.os.Looper
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
+import com.example.playlistmaker.Util.Create
+import com.example.playlistmaker.data.Track
+import com.example.playlistmaker.databinding.ActivityAudioPlayerBinding
+import com.example.playlistmaker.player.domain.models.PlayerState
 import com.google.gson.Gson
+import kotlinx.coroutines.Runnable
+import java.util.Locale
 
 class AudioPlayerActivity : AppCompatActivity() {
-    private lateinit var playerTime: TextView
-    private lateinit var track: Track
+
+    private var binding: ActivityAudioPlayerBinding? = null
+    private var track: Track? = null
     private var audioPlayerHolder: AudioPlayerHolder? = null
+    private var player = Create.providePlayer()
+    private var state = PlayerState.DEF_STATE
+    private var playerTime: TextView? = null
+
+    private val mainThreadHandler = Handler(Looper.getMainLooper())
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_audio_player)
+
+        binding = ActivityAudioPlayerBinding.inflate(layoutInflater)
+        setContentView(binding!!.root)
         playerTime = findViewById(R.id.player_time)
-        val backButton: Button = findViewById(R.id.buttonBackPlayer)
-        val addToPlayList: ImageView = findViewById(R.id.button_add_track)
-        val playPause: ImageView = findViewById(R.id.button_play)
-        val likeButton: ImageView = findViewById(R.id.button_like)
+
 
         audioPlayerHolder = AudioPlayerHolder(this)
```

Comment:
🍏 audioPlayerHolder используется только внутри этой функции, его можно сделать локальной неизменяемой переменной.
```suggestion
        val audioPlayerHolder = AudioPlayerHolder(this)
```